watches/util/ESClientProducer.py

```diff
@@ -0,0 +1,37 @@
+"""Provider of Elasticsearch client."""
+
+from elasticsearch import Elasticsearch
+
+def create_client(options):
+    """Produces a new Elasticsearch client according to provided options."""
+
+    user_kwargs = {}
+
+    if "--url" in options:
+        user_kwargs.update({
+            'hosts': options["--url"]
+        })
+
+    # "true" is not translated to native True value ?
+    if "--sniff" in options and "true" == options["--sniff"]:
+        user_kwargs.update({
+            # sniff before doing anything
+            "sniff_on_start": True,
+            # refresh nodes after a node fails to respond
+            "sniff_on_connection_fail": True,
+            # and also every 60 seconds
+            "sniffer_timeout": 60
+        })
+
+    # We can test just for --cacert because all the three options are required
+    # if at least one of them is provided.
```

Comment:
Not exactly.  You can provide cacert by itself to do https with username/password/token auth.  However, if you provide either cert or key, you must provide all 3 - you cannot have a cert without a cacert and key, you cannot have a key without a cacert ahd cert.